File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lrclyrics/dialog/ConfirmDialogFragment.java
package com.wa2c.android.medoly.plugin.action.lrclyrics.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.text.TextUtils;


/**
 * Confirm dialog.
 */
public class ConfirmDialogFragment extends AbstractDialogFragment {

    /** Title key. */
    private final static String ARG_TITLE = "TITLE";
    /** Message key. */
    private final static String ARG_MESSAGE = "MESSAGE";


    private final static String ARG_IS_BUTTON_DEFAULT = "IS_BUTTON_DEFAULT";
    /** Positive button key. */
    private final static String ARG_POSITIVE_BUTTON = "POSITIVE_BUTTON";
    /** Neutral button key. */
    private final static String ARG_NEUTRAL_BUTTON = "NEUTRAL_BUTTON";
    /** Negative button key. */
    private final static String ARG_NEGATIVE_BUTTON = "NEGATIVE_BUTTON";


    /**
     * Create dialog instance. (OK/Cancel button)
     * @param message Message.
     * @param title Title.
     * @return Dialog instance.
     */
    static public ConfirmDialogFragment newInstance(CharSequence message, CharSequence title) {
        ConfirmDialogFragment fragment = new ConfirmDialogFragment();
        Bundle args = new Bundle();
        args.putCharSequence(ARG_MESSAGE, message);
        args.putCharSequence(ARG_TITLE, title);
        args.putBoolean(ARG_IS_BUTTON_DEFAULT, true);
        fragment.setArguments(args);

        return fragment;
    }

    /**
     * Create dialog instance. (Custom button)
     * @param message Message.
     * @param title Title.
     * @param positiveButton Positive button label.
     * @param neutralButton  Neutral button label.
     * @param negativeButton Negative button label.
     * @return Dialog instance.
     */
    static public ConfirmDialogFragment newInstance(String message, String title, String positiveButton, String neutralButton, String negativeButton) {
        ConfirmDialogFragment fragment = new ConfirmDialogFragment();
        Bundle args = new Bundle();
        args.putCharSequence(ARG_MESSAGE, message);
        args.putCharSequence(ARG_TITLE, title);
        args.putString(ARG_POSITIVE_BUTTON, positiveButton);
        args.putString(ARG_NEUTRAL_BUTTON,  neutralButton);
        args.putString(ARG_NEGATIVE_BUTTON, negativeButton);
        args.putBoolean(ARG_IS_BUTTON_DEFAULT, false);
        fragment.setArguments(args);

        return fragment;
    }



    /**
     * onCreateDialog
     */
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        super.onCreateDialog(savedInstanceState);
        Bundle args = getArguments();

        // dialog builder
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setTitle(args.getCharSequence(ARG_TITLE));
        builder.setMessage(args.getCharSequence(ARG_MESSAGE));

        // button.
        if (args.getBoolean(ARG_IS_BUTTON_DEFAULT)) {
            builder.setPositiveButton(android.R.string.ok, clickListener);
            builder.setNegativeButton(android.R.string.cancel, clickListener);
        } else {
            // Positive button
            String positive = args.getString(ARG_POSITIVE_BUTTON);
            if (!TextUtils.isEmpty(positive)) {
                builder.setPositiveButton(positive, clickListener);
            }
            // Neutral button
            String neutral = args.getString(ARG_NEUTRAL_BUTTON);
            if (!TextUtils.isEmpty(neutral)) {
                builder.setNeutralButton(neutral, clickListener);
            }
            // Negative button
            String negative = args.getString(ARG_NEGATIVE_BUTTON);
            if (!TextUtils.isEmpty(negative)) {
                builder.setNegativeButton(negative, clickListener);
            }
        }

        return  builder.create();
    }

}

File: app/src/main/java/com/wa2c/android/medoly/plugin/action/lrclyrics/dialog/ConfirmDialogFragment.kt
package com.wa2c.android.medoly.plugin.action.lrclyrics.dialog

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog

/**
 * Confirmation dialog
 */
class ConfirmDialogFragment : AbstractDialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        super.onCreateDialog(savedInstanceState)
        val args = arguments!!

        // dialog build
        return AlertDialog.Builder(context).apply {
            setTitle(args.getCharSequence(ARG_TITLE))
            setMessage(args.getCharSequence(ARG_MESSAGE))

            // button
            if (args.getBoolean(ARG_IS_BUTTON_DEFAULT)) {
                setPositiveButton(android.R.string.ok, null)
                setNeutralButton(android.R.string.cancel, null)
            } else {
                // Positive button
                args.getString(ARG_POSITIVE_BUTTON)?.let {
                    if (it.isNotEmpty()) setPositiveButton(it, null)
                }
                // Neutral button
                args.getString(ARG_NEUTRAL_BUTTON)?.let {
                    if (it.isNotEmpty()) setNeutralButton(it, null)
                }
                // Negative button
                args.getString(ARG_NEGATIVE_BUTTON)?.let {
                    if (it.isNotEmpty()) setNegativeButton(it, null)
                }
            }
        }.create()
    }

    companion object {

        /** Title key.  */
        private const val ARG_TITLE = "TITLE"
        /** Message key. */
        private const val ARG_MESSAGE = "MESSAGE"


        private const val ARG_IS_BUTTON_DEFAULT = "IS_BUTTON_DEFAULT"
        /** Positive button key.  */
        private const val ARG_POSITIVE_BUTTON = "POSITIVE_BUTTON"
        /** Neutral button key.  */
        private const val ARG_NEUTRAL_BUTTON = "NEUTRAL_BUTTON"
        /** Negative button key.  */
        private const val ARG_NEGATIVE_BUTTON = "NEGATIVE_BUTTON"

        /**
         * Create dialog instance. (Customized button)
         * @param message A message.
         * @param title A title.
         * @param positiveButton Positive button. None if null.
         * @param neutralButton  Neutral button. None if null.
         * @param negativeButton Negative button. None if null.
         * @return New dialog instance.
         */
        fun newInstance(message: CharSequence, title: CharSequence?, positiveButton: String?, neutralButton: String?, negativeButton: String?): ConfirmDialogFragment {
            return ConfirmDialogFragment().apply {
                arguments = Bundle().apply {
                    putCharSequence(ARG_MESSAGE, message)
                    putCharSequence(ARG_TITLE, title)
                    putString(ARG_POSITIVE_BUTTON, positiveButton)
                    putString(ARG_NEUTRAL_BUTTON, neutralButton)
                    putString(ARG_NEGATIVE_BUTTON, negativeButton)
                    putBoolean(ARG_IS_BUTTON_DEFAULT, false)
                }
            }
        }
    }

}
